### PR TITLE
Add CAN library

### PR DESCRIPTION
**Pull request by jack-greenberg**
Adds CAN library for use in applications. Also includes:

* Legacy CAN library from `MKV-Code` that can be used during the transition so that none of the old code breaks
* A "mock" library that includes all of the API functions to be used in unit tests
* Unit tests that test the receive mock to make sure the receive filters work

The legacy library has a deprecation warning, and should eventually be removed once all the old applications are migrated over.

Should be merged after #13 - Integrate CMocka unit testing library

### CODE REVIEW

**Comment by awenstrup** [comment]
Curious if this is happening alongside Hwei Shin and Izzi's CAN work; is there a clear boundary of where this project ends and their's begins? Are the three of you (+ maybe Manu) having integration discussions?

**Comment by jack-greenberg** [comment]
Yeah this is more of a CAN driver that interacts with hardware. Izzi and Hwei Shin’s code is an app-level library that will invoke the functions of the driver. We had a conversation the other night and are synced up on progress :)

The two pieces are independent.

File: libs/can/api.h
Comment (awenstrup) [inline libs/can/api.h:15]: What's a CAN dlc?

Comment (awenstrup) [reply libs/can/api.h:15]: Nvm this was very googleable. In case anyone else is curious it's the data length code indicating the number of data bytes in the message.

**Review by awenstrup — COMMENTED** [review]
looks good to me! let me know if there's anything specific I should look at but the interface seems to make sense :)

**Comment by jack-greenberg** [comment]
Rx and tx both tested with a Canable device and found to be working. Interrupts not yet tested, but would like to merge before anyways--interrupts can come later.

This commit also introduces some Bazel fixes that I missed before having to do with link-time optimization in compiling and dependency issues with linker scripts.

**Comment by awenstrup** [comment]
> Rx and tx both tested with a Canable device and found to be working. Interrupts not yet tested, but would like to merge before anyways--interrupts can come later.
> 
> This commit also introduces some Bazel fixes that I missed before having to do with link-time optimization in compiling and dependency issues with linker scripts.

I can't approve as long as this is still a draft, but I'm happy to approve once that's changed.

**Review by awenstrup — APPROVED** [review]

File: libs/can/can.c
Comment (jack-greenberg) [reply libs/can/can.c:8]: http://wiki.olinelectricmotorsports.com/display/EL21/Git+Revamp+V2

I don't feel like we need to distinguish between public and private things. If a function is private to a specific translation unit (i.e. file), it can be private by just declaring it `static`. Or we can just prefix it with `private_`

**Pull request merged**
